std/algorithm/mutation.d

```diff
@@ -562,6 +562,17 @@ void fill(Range, Value)(Range range, Value value)
     }
 }
 
+import std.range : ElementEncodingType;
+import std.traits : isSomeChar, isMutable;
+
+/// ditto
+void fill(A, Value)(auto ref A array, auto ref Value value)
+if (isArray!A && isSomeChar!(ElementEncodingType!A) &&
```

Comment:
It must have `opIndex` with one parameter of type `ElementEncodingType!A`. I'm tempted to use a `is(typeof((){array[] = value}))` to avoid a heavy constraint...Would you have a better constraint as example ?